### ISSUE TITLE
relative path console.font no longer applied

### ISSUE THREAD

**Issue by ryantm** [NixOS/nixpkgs#130377]
I enabled the configuration
```
console = {
  font = "ter-i22b";
  packages = [ pkgs.terminus_font ];
};
```

but my console does not have this font. I get an error if I try to run the `systemd-vconsole-setup`:

```
Jul 16 07:16:02 usb-tester14 systemd[1]: Starting Setup Virtual Console...
Jul 16 07:16:02 usb-tester14 systemd-vconsole-setup[30864]: setfont: ERROR setfont.c:421 kfont_load_font: Unable to find file: ter-i22b
Jul 16 07:16:02 usb-tester14 systemd[1]: Finished Setup Virtual Console.
```

This worked in 20.03 but not now in 21.05.

If I specify the absolute path of the font like:

```
console = {
  font = "${pkgs.terminus_font}/share/consolefonts/ter-i22b.psf.gz";
};
```

I do not get that error. (I have not confirmed the font actually gets set since this is a remote computer I don't have immediate access to.)

My current theory is the `consoleEnv` is not available for `systemd-vconsole-setup`.

@rnhmjoj master of console.nix

**Comment by rnhmjoj** [comment]
Looks like you're right: consoleEnv is only used in the initrd. When `earlySetup` is true, the font is getting set, but the change is lost later when udev starts `systemd-vconsole-setup`. What's strange is that on my machines (with `earlySetup`) I never had this issue.

**Comment by jakobrs** [comment]
#125521

**Comment by rnhmjoj** [comment]
The fix is currently in staging and staging-21.05: we only have to wait for the next staging merge.
Credits to @zanculmarktum.

**Comment by rnhmjoj** [comment]
The fix reached both the stable and unstable channel.
